### DISCUSSION TITLE
Using tabs in Quarto

### DISCUSSION THREAD

**Discussion by Sidy2015** [quarto-dev/quarto-cli#9481] [Q&A] [answered]
### Description

Hi, i want to add tabs in Quarto. I will have 8 tabs, so i decide to make aloop in order to generate sections of text and print a dotplot under each tab.

I try the following code. The code must print dotplot under each tab

    ```r
    # Ligands-Receptors Couples (CellPhonDB)
    ::: {.panel-tabset .nav-pills}
    ```{r}
    #| label: dotplotcpdb
    #| fig.height: 10
    #| fig.width: 18
    #| results: 'asis'
     templateDtpcpdb <- "## %s
     Below are displayed the most significant interactions for the %s
     "
     for (i in 1:length(Fatlaslianacpdb)){
       cat('\n')
       cat(sprintf(templateDtpcpdb, unique(Fatlas.seu[[i]]$Sample), unique(Fatlas.seu[[i]]$Sample)))
       print(dotplotliana(i, Fatlaslianacpdb, Fatlaslianacpdb.sign))
      cat('\n')
    }```
    :::
    ```    

 

Nothing is displayed when using this code. Can you please tell me where there is an error?
Thank you for your help

**Comment by mcanouil** [comment, marked as answer]
Your code is not valid:
1. Code cells delimitation is wrong as you have backticks on the same line as brackets, see <https://quarto.org/docs/authoring/markdown-basics.html#source-code>.
2. Your are using Rmkarkdown option instead of Quarto options, see <https://quarto.org/docs/reference/cells/cells-knitr.html>.
3. You have included a level one header in your tabset div, see <https://quarto.org/docs/output-formats/html-basics.html#tabsets>.
4. And your code is not reproducible so we can't use it. We don't have your data, see <https://quarto.org/bug-reports.html#small-is-beautiful-aim-for-a-single-document-with-10-lines>.

````qmd
---
format: html
---

Below are tables of the first six rows of the Palmer penguins dataset by species:

::: {.panel-tabset}

```{r}
#| echo: false
#| output: asis
library(palmerpenguins)
cat(
  sprintf(
    "- `%s` (@tbl-%s)",
    levels(penguins[["species"]]),
    levels(penguins[["species"]])
  ),
  sep = "\n"
)

for (ispecies in levels(penguins[["species"]])) {
  tab <- penguins |>
    dplyr::filter(species %in% ispecies) |>
    dplyr::select(-species) |>
    head() |>
    knitr::kable()

  cat(sprintf("\n## %s\n", ispecies))
  print(tab)
}
```

:::

````


More:

- https://mickael.canouil.fr/posts/2023-03-05-quarto-auto-table-crossref/
- https://bookdown.org/yihui/rmarkdown-cookbook/child-document.html
- https://bookdown.org/yihui/rmarkdown-cookbook/knit-expand.html
- https://bookdown.org/yihui/rmarkdown-cookbook/results-asis.html
- https://github.com/quarto-dev/quarto-examples

**Reply by Sidy2015** [reply]
Thank you @mcanouil for your reply and the ressources! It works now

**Reply by harrelfe** [reply]
To do this for a wider variety of per-tab output objects see the R `qreport` package `maketabs` function.
